src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BzlmodRepoRuleCreator.java

```diff
@@ -93,4 +99,51 @@ public static Rule createRule(
     }
     return rule;
   }
+
+  public static void validateLabelAttrs(Rule rule, ModuleExtensionId owningExtension)
+      throws EvalException {
+    for (var attribute : rule.getRuleClassObject().getAttributes()) {
+      Object value = rule.getAttr(attribute.getName());
+      Collection<?> toValidate =
+          switch (value) {
+            case List<?> list -> list;
+            case Map<?, ?> map -> map.keySet();
+            case null, default -> singletonList(value);
+          };
+      for (var item : toValidate) {
+        if (!(item instanceof Label label)) {
+          continue;
+        }
+        if (label.getRepository().isVisible()) {
+          continue;
+        }
+        String unprefixedRepoName = label.getName().substring(label.getName().lastIndexOf('~') + 1);
```

Comment:
`label.getName()` is the target name. This line is essentially the same as `String .. = label.getName();` if the target name doesn't contain a `~`. Just use `label.getRepository().getName()`